src/index.ts

```diff
@@ -0,0 +1,759 @@
+      return schoolId != null;
+    });
+
+  const middleSchools: [string, SchoolType][] = dom
+    .querySelectorAll("#schoolCont #middleSchool td a")
+    .map((schoolElement): [string, SchoolType] => {
+      const href = schoolElement.getAttribute("href");
+      if (href) {
+        return [new URLSearchParams(href).get("S"), "Middle"];
+      }
+      return undefined;
+    })
+    .filter((schoolId) => {
+      return schoolId != null;
+    });
+
+  let schools: [string, SchoolType][] = [];
+
+  if (schoolType === "High") schools = highSchools;
+  else if (schoolType === "Middle") schools = middleSchools;
+  else {
+    schools = highSchools.concat(middleSchools);
+  }
+
+  const promises: Promise<School>[] = schools.map((school) => {
+    return new Promise((resolve) => {
+      getSchoolWithType(districtId, school[0], school[1]).then(
+        (asSchoolType) => {
+          resolve({
+            districtId,
+            id: school[0],
+            name: asSchoolType.name,
+            schoolType: school[1],
+            sports: asSchoolType.sports,
+          });
+        }
+      );
+    });
+  });
+
+  return await Promise.all(promises);
+};
+
+export const getSchoolIds = async (
+  districtId: string,
+  schoolType?: SchoolType
+): Promise<School["id"][]> => {
+  const district = await axios.get(
+    "https://www.rankonesport.com/Schedules/View_Schedule_All_Web.aspx",
+    {
+      params: {
+        D: districtId,
+        MT: 0,
+      },
+    }
+  );
+
+  const dom = parse(district.data);
+
+  const highSchools = dom
+    .querySelectorAll("#schoolCont #highCont td a")
+    .map((schoolElement): string => {
+      const href = schoolElement.getAttribute("href");
+      if (href) {
+        return new URLSearchParams(href).get("S");
+      }
+      return undefined;
+    })
+    .filter((schoolId) => {
+      return schoolId != null;
+    });
+
+  const middleSchools = dom
+    .querySelectorAll("#schoolCont #middleSchool td a")
+    .map((schoolElement): string => {
+      const href = schoolElement.getAttribute("href");
+      if (href) {
+        return new URLSearchParams(href).get("S");
+      }
+      return undefined;
+    })
+    .filter((schoolId) => {
+      return schoolId != null;
+    });
+
+  let schools: string[] = [];
+
+  if (schoolType === "High") schools = highSchools;
+  else if (schoolType === "Middle") schools = middleSchools;
+  else {
+    schools = highSchools.concat(middleSchools);
+  }
+
+  return schools;
+};
+
+export const getSchoolWithType = async (
+  districtId: string,
+  schoolId: string,
+  schoolType: SchoolType
+): Promise<School> => {
+  const school = await axios.get(
+    "https://www.rankone.com/Schedules/View_Schedule_All_Web.aspx",
+    {
+      params: {
+        P: "0",
+        D: districtId,
+        S: schoolId,
+        Mt: "0",
+      },
+    }
+  );
+
+  const dom = parse(school.data);
+
+  const schoolName = dom.querySelector("#lbl_PageTitle")?.innerText ?? "";
```

Comment:
would null be more appropriate here rather than a blank string?

Reply:
the types you made don't allow `null`

Reply:
oh, would it be worth fixing that, or leaving it a string

Reply:
thats what I did, it's left as `""`

Reply:
resolved on discord